contracts/token/TransferManager.sol

```diff
@@ -264,7 +251,7 @@ contract TransferManager is Pausable, ERC20 {
         require(identityRegistry.isVerified(_to), "Identity is not verified.");
 
         _mint(_to, _amount);
-        updateShareholders(_to);
+        compliance.created(_to, _amount);
```

Comment:
Compliance should be checked before minting to revert in cases where compliance will reject the transfers

Reply:
yes, it should be like this, but rather in `mint()` than in `_mint()` in my opinion

Reply:
_mint() is a standard function we have derived from ERC20 like _transfer and other methods. Imo we should include TREX specific methods here.